packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -45,73 +67,82 @@ contract RepPriceOracle is IRepPriceOracle, Initializable {
+        uint256 _price = (_repPriceCumulative - _exchangeData.repPriceAccumulated) * 10**18 / _blocksElapsed / Q112;
+        require(_price > 0, "Price should not be 0");
 
-        return _exchangeData;
-    }
+        uint256 _secondsElapsed = _blockTimestamp - _exchangeData.blockTimestamp;
+        uint256 _priceAverage = _price;
 
-    function getWeight(uint256 _blockDelta) public view returns (uint256) {
-        uint256 _weight = _blockDelta * 1 ether / tau;
-        if (_weight > 1 ether) {
-            return 1 ether;
+        if (_secondsElapsed < period) {
+            _priceAverage = (_exchangeData.price * (period - _secondsElapsed) + _price * _secondsElapsed) / period;
         }
-        return _weight;
+
+        _exchangeData.blockNumber = _blockNumber;
+        _exchangeData.blockTimestamp = _blockTimestamp;
+        _exchangeData.repPriceAccumulated = _repPriceCumulative;
+        _exchangeData.price = _priceAverage;
+        return _exchangeData;
     }
 
     function initializeUniverse(IV2ReputationToken _reputationToken) private {
-        exchangeData[address(_reputationToken)].exchange = getOrCreateUniswapExchange(_reputationToken);
+        uint256 _blockNumber = block.number;
+        IUniswapV2 _exchange = getOrCreateUniswapExchange(_reputationToken);
+        exchangeData[address(_reputationToken)].exchange = _exchange;
+        if (_blockNumber != _exchange.blockNumberLast()) {
+            _exchange.sync();
+        }
         uint256 _initialPrice = getInitialPrice(_reputationToken);
         exchangeData[address(_reputationToken)].price = _initialPrice;
-        exchangeData[address(_reputationToken)].blockNumber = block.number;
+        exchangeData[address(_reputationToken)].blockNumber = _blockNumber;
+        exchangeData[address(_reputationToken)].blockTimestamp = block.timestamp; // solium-disable-line security/no-block-members
+        (address token0, address token1) = uniswapFactory.sortTokens(cash, address(_reputationToken));
+        bool repIsToken0 = token0 == address(_reputationToken);
+        exchangeData[address(_reputationToken)].repIsToken0 = repIsToken0;
+        exchangeData[address(_reputationToken)].repPriceAccumulated = repIsToken0 ? _exchange.price0CumulativeLast() : _exchange.price1CumulativeLast();
     }
 
     function getInitialPrice(IV2ReputationToken _reputationToken) private view returns (uint256) {
         IUniverse _parentUniverse = _reputationToken.getUniverse().getParentUniverse();
-        uint256 _initialPrice = genesisInitialRepPriceinAttoCash;
         if (_parentUniverse != IUniverse(0)) {
             IV2ReputationToken _parentReputationToken = _parentUniverse.getReputationToken();
-            _initialPrice = exchangeData[address(_parentReputationToken)].price;
+            return exchangeData[address(_parentReputationToken)].price;
         }
-        return _initialPrice;
+        return genesisInitialRepPriceinAttoCash;
     }
```

Comment:
Consider inverting the conditional into a guard clause:
```solidity
IUniverse _parentUniverse = _reputationToken.getUniverse().getParentUniverse();
if (_parentUniverse == IUniverse(0)) return genesisInitialRepPriceinAttoCash;
IV2ReputationToken _parentReputationToken = _parentUniverse.getReputationToken();
_initialPrice = exchangeData[address(_parentReputationToken)].price;
return exchangeData[address(_parentReputationToken)].price;

Reply:
Reasoning: The "interesting" part of a function shouldn't live inside of a conditional.  Conditional contents should (ideally) contain the uninteresting or simplest path while the main path is the most interesting one.